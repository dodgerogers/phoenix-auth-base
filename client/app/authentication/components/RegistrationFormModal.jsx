import React from 'react'
import PropTypes from 'prop-types';
import AuthenticationModal from './AuthenticationModal';
import RegistrationFormContainer from '../containers/RegistrationFormContainer'
import AuthenticationLinks, { links } from './AuthenticationLinks';


const RegistrationFormModal = (props) => {
  return (
    <AuthenticationModal title="Create an account" {...props}>
      <RegistrationFormContainer {...props} />
      <AuthenticationLinks exclude={[links.REGISTRATION]} />
    </AuthenticationModal>
  );
}

export default RegistrationFormModal;
