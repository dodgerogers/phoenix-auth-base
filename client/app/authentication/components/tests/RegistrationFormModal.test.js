import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RegistrationFormModal from '../RegistrationFormModal';


describe('RegistrationForm', () => {
  const fullProps = {
    show: true,
    close: jest.fn(),
    onSubmit: jest.fn(),
  };

  const pureComponent = (props) => shallow(
    <RegistrationFormModal {...props} />
  );

  it('matches component snapshot', () => {
    const wrapper = pureComponent(fullProps);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
