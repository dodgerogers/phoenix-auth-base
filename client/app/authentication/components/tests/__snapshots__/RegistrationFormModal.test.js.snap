// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm matches component snapshot 1`] = `
<AuthenticationModal
  close={[Function]}
  onSubmit={[Function]}
  show={true}
  title="Create an account"
>
  <Connect(ReduxForm)
    close={[Function]}
    onSubmit={[Function]}
    show={true}
  />
  <AuthenticationLinks
    exclude={
      Array [
        "REGISTRATION",
      ]
    }
  />
</AuthenticationModal>
`;
